### DISCUSSION TITLE
how to get the Logic and arithmetic instructions of the PTX-level

### DISCUSSION THREAD

**Discussion by MWHYNOT** [accel-sim/accel-sim-framework#180] [General] [answered]
Excuse me，I want to collect all types of the PTX-level Logic and arithmetic instructions that a program may use when running on the simulator,like dp mul and the the number of bits that can be supported,can you tell me how to achieve it,thanks in advance

**Comment by JRPan** [comment, marked as answer]
Is this what you are looking for?
https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#instruction-set
Or you can check this file
https://github.com/accel-sim/gpgpu-sim_distribution/blob/dev/src/cuda-sim/ptx.l
and this
https://github.com/accel-sim/gpgpu-sim_distribution/blob/dev/src/cuda-sim/opcodes.def

**Reply by MWHYNOT** [reply]
Thank you very much！and do you know what is the meaning of " BB64"   " FF64"  "BB128" in the 
https://github.com/accel-sim/gpgpu-sim_distribution/blob/dev/src/cuda-sim/ptx.l
i look up the ptx8.1 document ，but not find their meangings

**Reply by JRPan** [reply]
http://gpgpu-sim.org/manual/index.php/Main_Page#PTXPlus_Implementation
.ff64 = PTXPlus version of 64 bit floating point number
.bb64 = PTXPlus version of 64 bit untyped
.bb128 = PTXPlus version of 128 bit untyped

**Reply by MWHYNOT** [reply]
it's so kind of you，Thank you so much for your help!
